### PR TITLE
User query

### PR DESCRIPTION
**Pull request by helloeduardo**
### Description

This PR adds models for orders and orderProducts.

This gets the user query up and running and fulfills the contract in #49 

### Testing
Testing is failing for the user query, but works in development
Will fix in a future PR

### Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [x] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### Checklist
- [x] I have written tests for this code (happy & sad)
- [ ] I have updated the Readme
- [ ] I ran full test suite - all tests passing

### CODE REVIEW

File: code/api/src/modules/order/model.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    status: {
+      type: DataTypes.STRING
+    }
+  })
+
+  Order.associate = function(models) {
+    Order.belongsTo(models.Subscription)
+    Order.hasMany(models.OrderProduct)
+    // Order.belongsToMany(models.User, {through: models.Subscription})
+    // Order.belongsToMany(models.Product, {through: models.OrderProduct})
```
Comment (Jonathan-M-Wilson) [inline code/api/src/modules/order/model.js]: We could probably remove this commented out code now that we figured out the nested issue we were running into earlier. :)

File: code/api/src/modules/order/resolvers.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+export async function getAll() {
+  return await models.Order.findAll({
+    include: [
+      { model: models.Subscription, as: 'subscription' }
+    ]
+  })
+}
+
+// Create order
+// export async function create(parentValue, { crateId }, { auth }) {
```
Comment (Jonathan-M-Wilson) [inline code/api/src/modules/order/resolvers.js]: Same here for code cleanup :)

File: code/api/src/modules/orderProduct/resolvers.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    include: [
+      { model: models.Product, as: 'product' },
+      { model: models.Order, as: 'order' }
+    ]
+  })
+}
+
+// Create order
+// export async function create(parentValue, { crateId }, { auth }) {
+//   if(auth.user && auth.user.id > 0) {
```
Comment (Jonathan-M-Wilson) [inline code/api/src/modules/orderProduct/resolvers.js]: Housekeeping: Code cleanup 🧹

File: code/api/src/modules/user/model.js [hunk 4]
```diff
@@ -40,6 +40,8 @@ module.exports = function(sequelize, DataTypes) {
 
   User.associate = function(models) {
     User.hasMany(models.Subscription)
+    // User.hasMany(models.Order, {through: models.Subscription})
```
Comment (Jonathan-M-Wilson) [inline code/api/src/modules/user/model.js]: Housekeeping: Code cleanup 🧹

File: code/api/src/modules/user/mutations.js [hunk 5]
```diff
@@ -49,6 +49,11 @@ export const userUpdate = {
       type: GraphQLInt
     },
 
+    name: {
```
Comment (Jonathan-M-Wilson) [inline code/api/src/modules/user/mutations.js:52]: Thank you for adding this! This was something I missed originally!

File: code/api/src/modules/user/tests/userQuery.test.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  // let order1;
+  // let orderProduct1;
+  // let product1;
+
+  beforeAll(async () => {
+
+    user1 = await models.User.create(mockData.userData1);
+    // crate1 = await models.Crate.create(mockData.crateData1);
+    // subscription1 = await models.Subscription.create(mockData.subscriptionData1);
+    // order1 = await models.Order.create(mockData.orderData1);
```
Comment (Jonathan-M-Wilson) [inline code/api/src/modules/user/tests/userQuery.test.js:31]: Housekeeping: Code cleanup 🧹

**Review by Todd-Estes — APPROVED** [review]

File: code/api/src/setup/schema/mutations.js [hunk 7]
```diff
@@ -6,6 +6,8 @@ import * as user from '../../modules/user/mutations'
 import * as product from '../../modules/product/mutations'
 import * as crate from '../../modules/crate/mutations'
 import * as subscription from '../../modules/subscription/mutations'
+import * as order from '../../modules/order/mutations'
+// import * as orderProduct from '../../modules/orderProduct/mutations'
```
Comment (Jonathan-M-Wilson) [inline code/api/src/setup/schema/mutations.js:10]: Housekeeping: Code cleanup 🧹

**Review by Jonathan-M-Wilson — CHANGES_REQUESTED** [review]
Looks great! My only recommendation is to go back and remove all the code that has been commented out and do some housekeeping. Great work!  :)

**Comment by Todd-Estes** [comment]
Great work! Just have to get the testing figured out!

**Review by Jonathan-M-Wilson — APPROVED** [review]
Perfect! Thank you!

**Review by araspaas — APPROVED** [review]

**Pull request merged**